orchestration/src/main/java/com/sap/ai/sdk/orchestration/GeneralizedFallbackDeserializer.java

```diff
@@ -0,0 +1,73 @@
+package com.sap.ai.sdk.orchestration;
+
+import com.fasterxml.jackson.annotation.JsonSubTypes;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.*;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import java.io.IOException;
+
+/**
+ * A deserializer that attempts to deserialize a JSON object into one of the subtypes of a base
+ * type.
+ *
+ * <p>This deserializer is useful when the exact subtype of an object is not known in advance, but
+ * the possible subtypes are known and annotated with {@link JsonSubTypes}.
+ *
+ * <p>When deserializing, this deserializer will attempt to deserialize the JSON object into each of
+ * the possible subtypes in turn. The first successful deserialization will be returned. If none of
+ * the deserializations are successful, a {@link JsonMappingException} will be thrown.
+ *
+ * @param <T> The base type to deserialize into.
+ */
+public class GeneralizedFallbackDeserializer<T> extends StdDeserializer<T> {
```

Comment:
(Question)

In what cases do we need this class? I would've expected this was default Jackson behavior 🤔

Reply:
I have removed `GeneralizedFallbackDeserializer`.

Jackson deserialiser resolves for sub type based on 
1. existence of any unique property by name 
2. values of some specified string property

For `LLMModuleResult` oneof interface, neither approaches can help resolve to one of the concrete classes, as the only discriminating information is the type of one of their properties. Similar cases seems highly likely unless OpenAPI spec definition takes this into account.